tests/unit/test_network_connections.py

```diff
@@ -3312,6 +3312,114 @@ def test_dns_options_argvalidator(self):
+                            "family": socket.AF_INET,
+                        }
+                    ],
+                    "route_append_only": False,
+                    "rule_append_only": False,
+                    "route": [],
+                    "route_metric6": None,
+                    "dhcp4_send_hostname": None,
+                    "dns": [
+                        {
+                            "prefix": 32,
+                            "address": "2001:db8::20",
+                            "family": socket.AF_INET6,
+                        }
+                    ],
+                    "dns_options": [],
+                    "dns_search": [],
+                },
+                "mac": None,
+                "controller": None,
+                "ieee802_1x": None,
+                "wireless": None,
+                "mtu": None,
+                "parent": None,
+                "port_type": None,
+                "zone": None,
+            }
+        ]
+
+        input_connection2 = [
+            {
+                "name": "ethtest2",
+                "type": "ethernet",
+                "ip": {
+                    "gateway6": "2001:db8::1",
+                    "gateway4": None,
+                    "route_metric4": None,
+                    "auto6": False,
+                    "ipv6_disabled": False,
+                    "dhcp4": False,
+                    "address": [
+                        {
+                            "prefix": 32,
+                            "address": "2001:db8::4",
+                            "family": socket.AF_INET6,
+                        }
+                    ],
+                    "route_append_only": False,
+                    "rule_append_only": False,
+                    "route": [],
+                    "route_metric6": None,
+                    "dhcp4_send_hostname": None,
+                    "dns": [
+                        {
+                            "prefix": 24,
+                            "address": "192.168.174.1",
+                            "family": socket.AF_INET,
+                        }
+                    ],
+                    "dns_options": [],
+                    "dns_search": [],
+                },
+                "mac": None,
+                "controller": None,
+                "ieee802_1x": None,
+                "wireless": None,
+                "mtu": None,
+                "parent": None,
+                "port_type": None,
+                "zone": None,
+            }
+        ]
+        validator = network_lsr.argument_validator.ArgValidator_ListConnections()
+        self.assertRaises(
+            ValidationError,
+            validator.validate_connection_one,
+            "nm",
+            input_connection1,
+            0,
+        )
+        self.assertRaises(
+            ValidationError,
+            validator.validate_connection_one,
+            "nm",
+            input_connection2,
+            0,
+        )
```

Comment:
These should be two test cases, one for IPv4 and one for IPv6

Reply:
Created separated test cases:
`def test_ipv6_dns_without_ipv6_config(self)`
`def test_ipv4_dns_without_ipv4_config(self)`